### DISCUSSION TITLE
Why does `rsa::Keypair` hold `Arc<RsaKeyPair>` while others doesn't?

### DISCUSSION THREAD

**Discussion by drHuangMHT** [libp2p/rust-libp2p#3678] [Q&A] [answered]
As the title suggest. This particular keypair is essentially holding a reference that is not mutable. Why? If it's performance-related, why don't other types of keypair do the same?

**Comment by thomaseizinger** [comment, marked as answer]
It is wrapped in an `Arc` to make the `Keypair` type clonable.

**Reply by drHuangMHT** [reply]
Oh the keypair provided by `ring` does not implement `Clone`. I can't come up with a reason why though.
